tesseract_common/include/tesseract_common/utils.h

```diff
@@ -368,5 +368,30 @@ inline bool almostEqualRelativeAndAbs(double a,
   return (diff <= largest * max_rel_diff);
 }
 
+/**
+ * @brief Check if two Eigen VectorXd are relatively equal
+ * @details This is a vectorized implementation of the function above.
+ * @param a A vector of double
+ * @param b A vector of double
+ * @param max_diff The max diff when comparing std::abs(a - b) <= max_diff, if true they are considered equal
+ * @param max_rel_diff The max relative diff std::abs(a - b) <= largest * max_rel_diff, if true considered equal. The
+ * largest is the largest of the absolute values of a and b.
+ * @return True if they are relativily equal, otherwise false.
+ */
+inline bool almostEqualRelativeAndAbs(const Eigen::Ref<const Eigen::VectorXd>& a,
+                                      const Eigen::Ref<const Eigen::VectorXd>& b,
+                                      double max_diff,
+                                      double max_rel_diff = std::numeric_limits<double>::epsilon())
+{
+  // Check if the numbers are really close -- needed
+  // when comparing numbers near zero.
+  Eigen::VectorXd diff_abs = (a - b).cwiseAbs();
+  double diff = diff_abs.maxCoeff();
+  if (diff <= max_diff)
+    return true;
+
+  return (diff_abs.array() <= (max_rel_diff * a.cwiseAbs().cwiseMax(b.cwiseAbs())).array()).all();
```

Comment:
These element-wise operations would probably be a little cleaner if you worked with `Eigen::ArrayXd` instead of `Eigen::VectorXd`

Reply:
I will see what I can come up with.